osmnx/pois.py

```diff
@@ -76,32 +157,51 @@ def osm_poi_download(polygon=None, amenities=None, north=None, south=None, east=
     ----------
     poly : shapely.geometry.Polygon
         Polygon that will be used to limit the POI search.
+    north, south, east, west: float
+        Bounding box used to limit the POI search.
     amenities : list
-        List of amenities that will be used for finding the POIs from the selected area.
+        List of amenities that will be used for finding the POIs from the
+        selected area. Results returned are the _union_, not _intersection_ of
+        each individual amenity tag; each result matches at least one tag given.
+        Equivalent to `tags={'amenity': amenities}`. Some POIs don't use the
+        `amenity` key, in which case you should use the `tags` option
+    tags : dict[str, Union[bool, str, List[str]]]
+        Dict of tags used for finding POIs from the selected area. Results
+        returned are the _union_, not _intersection_ of each individual tag;
+        each result matches at least one tag given.
+
+        The keys should be the OSM tag key, (i.e. `amenity`, `landuse`,
+        `highway`, etc) and the values should be `True` to retrieve all objects
+        with the given tag, a string to get a single tag-value combination, and
+        a list of strings to get multiple values for the given tag. For example,
+
+            tags = {
+                'amenity':True,
+                'landuse':['retail','commercial'],
+                'highway':'bus_stop'
+            }
+
+        Would return all amenities, `landuse=retail` and `landuse=commercial`,
+        and `highway=bus_stop`. If both `amenities` and `tags` are combined, any
+        `amenity` tags will be combined.
+    timeout : int
+        Timeout for the API request.
 
     Returns
     -------
-    gdf : geopandas.GeoDataFrame
-        Points of interest and the tags associated with them as geopandas GeoDataFrame.
+    responses : dict
+        JSON response with POIs from Overpass API server.
     """
 
-    if polygon:
-        # Bounds
+    if polygon is not None:
         west, south, east, north = polygon.bounds
 
-        # Parse the Overpass QL query
-        query = parse_poi_query(amenities=amenities, west=west, south=south, east=east, north=north)
-
-    elif not (north is None or south is None or east is None or west is None):
-        # TODO: Add functionality for subdividing search area geometry based on max_query_area_size
-        # Parse Polygon from bbox
-        #polygon = bbox_to_poly(north=north, south=south, east=east, west=west)
+    elif not all(isinstance(x, float) for x in [north, south, east, west]):
```

Comment:
Will this fail if the passed in bounding box contains integer coordinates? They should be valid too.

Reply:
Yes, good catch.